### ISSUE TITLE
completing crashes elvish

### ISSUE THREAD

**Issue by huiyiqun** [elves/elvish#253]
```
~> ls Downloads/

runtime error: integer divide by zero
goroutine 1 [running]:
github.com/elves/elvish/sys.DumpStack(0xc42004ce00, 0x1)
        /tmp/yaourt-tmp-huiyiqun/aur-elvish-git/src/build/src/github.com/elves/elvish/sys/dumpstack.go:7 +0x79
github.com/elves/elvish/run.rescue()
        /tmp/yaourt-tmp-huiyiqun/aur-elvish-git/src/build/src/github.com/elves/elvish/run/run.go:115 +0xd2
panic(0x68cb20, 0xc42000c100)
        /usr/lib/go/src/runtime/panic.go:458 +0x243
github.com/elves/elvish/edit.(*completion).List(0xc42011a170, 0x75, 0x1d, 0xc42004d258)
        /tmp/yaourt-tmp-huiyiqun/aur-elvish-git/src/build/src/github.com/elves/elvish/edit/completion.go:290 +0xae9
github.com/elves/elvish/edit.(*writer).refresh(0xc4200e0290, 0xc42011a0a8, 0x0, 0xc4202ac0f8, 0x0)
        /tmp/yaourt-tmp-huiyiqun/aur-elvish-git/src/build/src/github.com/elves/elvish/edit/writer.go:515 +0x1189
github.com/elves/elvish/edit.(*Editor).refresh(0xc42011a000, 0x7f3b3c060100, 0xc4200d4420, 0xc4202ac0f8)
        /tmp/yaourt-tmp-huiyiqun/aur-elvish-git/src/build/src/github.com/elves/elvish/edit/editor.go:182 +0x3ad
github.com/elves/elvish/edit.(*Editor).ReadLine(0xc42011a000, 0x0, 0x0, 0x0, 0x0)
        /tmp/yaourt-tmp-huiyiqun/aur-elvish-git/src/build/src/github.com/elves/elvish/edit/editor.go:342 +0x4d9
github.com/elves/elvish/run.interact.func1(0x694f80, 0xc420174d30, 0x1d, 0x6c5f01)
        /tmp/yaourt-tmp-huiyiqun/aur-elvish-git/src/build/src/github.com/elves/elvish/run/run.go:171 +0x2a
github.com/elves/elvish/run.interact(0xc4200d8120, 0xc4200d40c0)
        /tmp/yaourt-tmp-huiyiqun/aur-elvish-git/src/build/src/github.com/elves/elvish/run/run.go:182 +0x1e8
github.com/elves/elvish/run.Main()
        /tmp/yaourt-tmp-huiyiqun/aur-elvish-git/src/build/src/github.com/elves/elvish/run/run.go:106 +0x23f
main.main()
        /tmp/yaourt-tmp-huiyiqun/aur-elvish-git/src/build/src/github.com/elves/elvish/main.go:8 +0x14
goroutine 17 [syscall, 1 minutes, locked to thread]:
runtime.goexit()
        /usr/lib/go/src/runtime/asm_amd64.s:2086 +0x1

goroutine 5 [syscall]:
os/signal.signal_recv(0x9ed240)
        /usr/lib/go/src/runtime/sigqueue.go:116 +0x157
os/signal.loop()
        /usr/lib/go/src/os/signal/signal_unix.go:22 +0x22
created by os/signal.init.1
        /usr/lib/go/src/os/signal/signal_unix.go:28 +0x41

goroutine 6 [select, 1 minutes, locked to thread]:
runtime.gopark(0x6e4508, 0x0, 0x6c5e5e, 0x6, 0x18, 0x2)
        /usr/lib/go/src/runtime/proc.go:259 +0x13a
runtime.selectgoImpl(0xc42002a730, 0x0, 0x18)
        /usr/lib/go/src/runtime/select.go:423 +0x11d9
runtime.selectgo(0xc42002a730)
        /usr/lib/go/src/runtime/select.go:238 +0x1c
runtime.ensureSigM.func1()
        /usr/lib/go/src/runtime/signal1_unix.go:304 +0x2f3
runtime.goexit()
        /usr/lib/go/src/runtime/asm_amd64.s:2086 +0x1

goroutine 7 [chan receive, 1 minutes]:
github.com/elves/elvish/run.handleHupAndQuit.func1(0xc420064240)
        /tmp/yaourt-tmp-huiyiqun/aur-elvish-git/src/build/src/github.com/elves/elvish/run/run.go:235 +0x65
created by github.com/elves/elvish/run.handleHupAndQuit
        /tmp/yaourt-tmp-huiyiqun/aur-elvish-git/src/build/src/github.com/elves/elvish/run/run.go:241 +0x148

goroutine 18 [chan receive]:
github.com/elves/elvish/run.logSignals.func1(0xc4200643c0)
        /tmp/yaourt-tmp-huiyiqun/aur-elvish-git/src/build/src/github.com/elves/elvish/run/run.go:225 +0x6a
created by github.com/elves/elvish/run.logSignals
        /tmp/yaourt-tmp-huiyiqun/aur-elvish-git/src/build/src/github.com/elves/elvish/run/run.go:228 +0x8a

goroutine 20 [syscall, 1 minutes]:
syscall.Syscall(0x0, 0x7, 0xc420104020, 0x1, 0x0, 0x0, 0x0)
        /usr/lib/go/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0x7, 0xc420104020, 0x1, 0x4, 0x0, 0x0, 0x0)
        /usr/lib/go/src/syscall/zsyscall_linux_amd64.go:783 +0x55
syscall.Read(0x7, 0xc420104020, 0x1, 0x4, 0x0, 0x0, 0x0)
        /usr/lib/go/src/syscall/syscall_unix.go:161 +0x49
os.(*File).read(0xc4200ea030, 0xc420104020, 0x1, 0x4, 0x0, 0x0, 0x0)
        /usr/lib/go/src/os/file_unix.go:228 +0x4f
os.(*File).Read(0xc4200ea030, 0xc420104020, 0x1, 0x4, 0x0, 0x0, 0x0)
        /usr/lib/go/src/os/file.go:101 +0x59
io.ReadAtLeast(0x9ebc00, 0xc4200ea030, 0xc420104020, 0x1, 0x4, 0x1, 0x0, 0x55, 0x0)
        /usr/lib/go/src/io/io.go:307 +0xa4
io.ReadFull(0x9ebc00, 0xc4200ea030, 0xc420104020, 0x1, 0x4, 0x6e24ca, 0xc41fff82fc, 0x0)
        /usr/lib/go/src/io/io.go:325 +0x58
fmt.(*readRune).readByte(0xc420104000, 0xc4200fa000, 0x60, 0x58)
        /usr/lib/go/src/fmt/scan.go:328 +0xc7
fmt.(*readRune).ReadRune(0xc420104000, 0x0, 0x28, 0x6e24c7, 0x6e24c6)
        /usr/lib/go/src/fmt/scan.go:344 +0x3a
fmt.(*ss).ReadRune(0xc4200fa000, 0x30, 0x28, 0x6abc20, 0x551701)
        /usr/lib/go/src/fmt/scan.go:189 +0x81
fmt.(*ss).getRune(0xc4200fa000, 0x1)
        /usr/lib/go/src/fmt/scan.go:211 +0x2f
fmt.(*ss).skipSpace(0xc4200fa000, 0x0)
        /usr/lib/go/src/fmt/scan.go:427 +0x2f
fmt.(*ss).SkipSpace(0xc4200fa000)
        /usr/lib/go/src/fmt/scan.go:305 +0x30
fmt.(*ss).doScanf(0xc4200fa000, 0x6c535b, 0x2, 0xc420038f00, 0x1, 0x1, 0x0, 0x0, 0x0)
        /usr/lib/go/src/fmt/scan.go:1176 +0x358
fmt.Fscanf(0x9ebc00, 0xc4200ea030, 0x6c535b, 0x2, 0xc420038f00, 0x1, 0x1, 0x0, 0x0, 0x0)
        /usr/lib/go/src/fmt/scan.go:143 +0xc5
github.com/elves/elvish/stub.relaySignals(0x9ebc00, 0xc4200ea030, 0xc4200dc1e0, 0xc4200dc240)
        /tmp/yaourt-tmp-huiyiqun/aur-elvish-git/src/build/src/github.com/elves/elvish/stub/stub.go:140 +0xf7
created by github.com/elves/elvish/stub.NewStub
        /tmp/yaourt-tmp-huiyiqun/aur-elvish-git/src/build/src/github.com/elves/elvish/stub/stub.go:73 +0x6e9

goroutine 21 [syscall, 1 minutes]:
syscall.Syscall6(0xf7, 0x1, 0x7618, 0xc420108000, 0x1000004, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/lib/go/src/syscall/asm_linux_amd64.s:44 +0x5
os.(*Process).blockUntilWaitable(0xc4200d8210, 0x0, 0x0, 0x0)
        /usr/lib/go/src/os/wait_waitid.go:28 +0xbc
os.(*Process).wait(0xc4200d8210, 0x0, 0x0, 0x0)
        /usr/lib/go/src/os/exec_unix.go:22 +0xab
os.(*Process).Wait(0xc4200d8210, 0x0, 0x0, 0x0)
        /usr/lib/go/src/os/doc.go:49 +0x2b
github.com/elves/elvish/stub.wait(0xc4200d8210, 0xc4200dc2a0)
        /tmp/yaourt-tmp-huiyiqun/aur-elvish-git/src/build/src/github.com/elves/elvish/stub/stub.go:159 +0x52
created by github.com/elves/elvish/stub.NewStub
        /tmp/yaourt-tmp-huiyiqun/aur-elvish-git/src/build/src/github.com/elves/elvish/stub/stub.go:74 +0x718

goroutine 22 [chan receive, 1 minutes]:
github.com/elves/elvish/run.Main.func2(0xc4200d8270)
        /tmp/yaourt-tmp-huiyiqun/aur-elvish-git/src/build/src/github.com/elves/elvish/run/run.go:92 +0x48
created by github.com/elves/elvish/run.Main
        /tmp/yaourt-tmp-huiyiqun/aur-elvish-git/src/build/src/github.com/elves/elvish/run/run.go:94 +0x198

execing recovery shell /bin/sh
```

**Comment by huiyiqun** [comment]
same issue when terminal has extremely low width.

**Comment by xiaq** [comment]
Fixed by fc9a56f350dd350611d53c94eac00e1041fae73b.
